File: slack_channel/wikipedia_event_handler.py
# coding=utf-8
from commands import WikipediaSearchCommand
from slack_channel.abstract_event_handler import AbstractEventHandler


command_symbol = "wik"

class WikipediaEventHandler(AbstractEventHandler):
    @property
    def command(self):
        return WikipediaSearchCommand()

    @property
    def name(self):
        return "Wikipedia"

    def get_usage(self):
        return self.command_trigger + "<query>"

    def can_handle(self, slack_event):
        text = slack_event["text"]
        return text.startswith(self.command_trigger)

    def _invoke_handler_logic(self, slack_event):
        q = slack_event["text"][len(self.command_trigger):]
        c = WikipediaSearchCommand()
        result = c.execute(q)
        if result is None:
            response_message = "No results found for \"" + q +  "\""
        else:
            response_message = result["text"] + " " + result["link"]
        self._send_message_response(response_message, slack_event)

    def _get_command_symbol(self):
        return command_symbol

File: bottomly.net/Bottomly/Slack/EventHandlers/WikipediaEventHandler.cs
using Bottomly.Commands;
using Bottomly.Configuration;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.Options;
using SlackNet.Events;

namespace Bottomly.Slack.EventHandlers;

public class WikipediaEventHandler(
    WikipediaSearchCommand command,
    ISlackMessageBroker broker,
    IOptions<BottomlyOptions> options,
    ILogger<WikipediaEventHandler> logger)
    : AbstractEventHandler(broker, options, logger)
{
    public override string Name => "Wikipedia";
    public override ICommand Command => command;
    protected override string CommandSymbol => "wik";
    protected override string GetUsage() => CommandTrigger + "<term>";

    protected override async Task InvokeHandlerLogicAsync(MessageEvent message)
    {
        var term = message.Text![CommandTrigger.Length..];
        var result = await command.ExecuteAsync(term);
        var response = result is null
            ? $"No results found for \"{term}\""
            : $"{result.Text} {result.Link}";
        await SendMessageResponseAsync(response, message);
    }
}